.github/workflows/foreman.yml

```
@@ -0,0 +1,67 @@
+---
+name: Foreman
+
+on:
+  - pull_request
+
+env:
+  RAILS_ENV: test
+  DATABASE_URL: postgresql://postgres:password@localhost/test
+  DATABASE_CLEANER_ALLOW_REMOTE_DATABASE_URL: true
+  BUNDLE_WITHOUT: "console:development:journald"
+
+jobs:
+  setup_matrix:
+    name: Setup matrix
+    runs-on: ubuntu-latest
+    outputs:
+      matrix: ${{ steps.build_matrix.outputs.matrix }}
+    steps:
+      - name: Build test matrix
+        id: build_matrix
+        uses: theforeman/gha-matrix-builder@v0
+  test:
+    name: "develop with Ruby ${{ matrix.ruby }} and Node ${{ matrix.node }}"
+    runs-on: ubuntu-latest
+    needs: setup_matrix
+    services:
+      postgres:
+        image: 'postgres:12'
+        ports: ['5432:5432']
+        options: --health-cmd pg_isready --health-interval 10s --health-timeout 5s --health-retries 5
+        env:
+          POSTGRES_PASSWORD: password
+    strategy:
+      fail-fast: false
+      matrix:
+        ruby: ${{ fromJson(needs.setup_matrix.outputs.matrix).ruby }}
+        node: ${{ fromJson(needs.setup_matrix.outputs.matrix).node }}
```

```suggestion
      matrix: ${{ fromJson(needs.setup_matrix.outputs.matrix) }}
```

This is simpler :)
(the individual ruby/node assignment was only needed for the "weird" matrix in the plugin test where we need to add more axis to the matrix, which doesn't work otherwise)

---

I was actually thinking this may be good if we also want to store the postgresql version in `matrix.json`. For when we update to > 12 (which we know is coming at some point).

---

which would mean this matrix won't include it and only test the default postgres, not sure this is "good"? :)

---

Fair point. I was thinking it may not be needed but if it's a single entry in the matrix it won't be a difference.